geniusyield-server-lib/src/GeniusYield/Server/Auth.hs

```diff
@@ -0,0 +1,36 @@
+module GeniusYield.Server.Auth (
+  ApiKey,
+  apiKeyFromText,
+  ApiKeyHeader,
+  apiKeyHeaderText,
+  apiKeyAuthHandler,
+) where
+
+import GHC.TypeLits (Symbol, symbolVal)
+import Network.Wai (Request (requestHeaders))
+import RIO
+import RIO.Text qualified as T
+import Servant
+import Servant.Server.Experimental.Auth (AuthHandler, mkAuthHandler)
+
+-- | The Api Key type.
+newtype ApiKey = ApiKey ByteString
+
+apiKeyFromText ∷ Text → ApiKey
+apiKeyFromText = ApiKey . encodeUtf8
+
+type ApiKeyHeader ∷ Symbol
+type ApiKeyHeader = "api-key"
+
+apiKeyHeaderText ∷ Text
+apiKeyHeaderText = symbolVal (Proxy ∷ Proxy ApiKeyHeader) & T.pack
+
+apiKeyAuthHandler ∷ ApiKey → AuthHandler Request ()
+apiKeyAuthHandler (ApiKey key) = mkAuthHandler handler
+ where
+  handler req = case lookup "api-key" (requestHeaders req) of
+    Nothing → throwError err401 {errBody = "Missing API key"}
```

Comment:
Could we make this error message a bit more descriptive?

```
Missing API key (please pass the api key in the api-key HTTP header) 
```

Or something like this could help the useres.

Reply:
Corresponding update has been made.